src/Card/Card.component.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+export type Props = {
+  /**
+   * Select Card Style
+   *
+   * @default 'primary'
+   **/
+  cardStyle: 'primary' | 'success' | 'default' | 'danger' | 'warning';
+  /**
+   * Select which color will be the background color
+   *
+   * @default '#FFFFFF'
+   **/
+  bgColor?: string;
+  /**
+   * Select which border of the Card is colored
+   *
+   * @default 'top'
+   **/
+  highlightBorder?: 'top' | 'left' | 'right' | 'bottom' | 'all';
+  /**
+   * From theme provider
+   *
+   * @default defaultTheme
+   **/
+  theme?: any;
+};
+
+const SCard = styled.div`
+  border-radius: ${(props: Props) => props.theme.card.borderRadius};
+  color: ${(props: Props) => props.theme.styles[props.cardStyle].cardColor};
+  font-family: ${(props: Props) => props.theme.typography.fontFamily};
+  font-size: ${(props: Props) => props.theme.card.fontSize};
+  padding: ${(props: Props) => props.theme.card.padding};
+  background: ${(props: Props) => (props.bgColor) ? props.bgColor : props.theme.card.background};
+  box-shadow: ${(props: Props) => props.theme.card.boxShadow};
+  border-style: solid;
+  border-width: 1px;
+
+  border-color: ${(props: Props) => (props.highlightBorder === 'all') ?
+  props.theme.styles[props.cardStyle].cardTopBorderColor :
+  props.theme.styles['default'].cardTopBorderColor};
+
+  border-${(props: Props) => (props.highlightBorder) ? props.highlightBorder : ''}-width: 5px;
```

Comment:
If there is no `highlightBorder`, this will render `border--width: 5px;`, which won't do anything but should be removed. Also, I believe `5px` should be defined in the theme. You may want to pull the condition out around the entire rule so you can write the style for `highlightBorder === 'all'`, `!highlightBorder`, and the other options separately.